packages/legacy/handlers/supergraph/src/index.ts

```diff
@@ -158,3 +159,24 @@ export default class SupergraphHandler implements MeshHandler {
     };
   }
 }
+
+function handleSupergraphResponse(
+  sdlOrIntrospection: string | DocumentNode,
+  interpolatedSource: string,
+) {
+  if (typeof sdlOrIntrospection === 'string') {
+    try {
+      return parse(sdlOrIntrospection, { noLocation: true });
+    } catch (e) {
+      throw new Error(
+        `Supergraph SDL must be a string, but got an invalid result from ${interpolatedSource} instead.\n Got result: ${sdlOrIntrospection}\n Got error: ${e.message}`,
```

Comment:
```suggestion
        `Supergraph SDL must be a valid schema string, but got an invalid result from ${interpolatedSource} instead.\n Got result: ${sdlOrIntrospection}\n Got error: ${e.message}`,
```

Reply:
How about now?

Reply:
It's better :-)
You think it's better to have the same message in the 2 cases ? We know in the first one that it is a string, so it's very pobable that there is just a syntaxe error in the schema.

Reply:
It returns the syntax error if it is a string but still I thought the endpoint might return random string even if it is supposed to return an object or other way around. So I think it'd better for the user to know it accepts both an object or a string.